tools/debug/dat.py

```diff
@@ -0,0 +1,198 @@
+from time import time
+import tkinter as tk
+from tkinter import *
+from tkinter import ttk
+import os
+import serial
+from pynput.keyboard import Key,Controller
+from tkinter.scrolledtext import ScrolledText
+from tkinter import messagebox
+from serial.serialutil import SerialException
+
+def usage():
+    print('*************************************************************')
+    print('\nUsage: %s <device port>' % sys.argv[0])
+    print('')
+    print('syntax :')
+    print('--------')
+    print('python3 %s <device port>' % sys.argv[0])
+    print('')
+    print('Below example if you give device port as argument: ')
+    print('--------------------------------------------')
+    print('python3 %s /dev/ttyUSB1' % sys.argv[0])
+    print('--------------------------------------------\n')
+    print('*************************************************************')
+
+#function use to setup the port Connection
+def setupPort(userport):
+    try:
+        PORT = serial.Serial(userport, baudrate=115200, timeout=0.1)
+        return PORT
+    except SerialException:
+        ErrorMsg = f"Failed to estabish UART connection for PORT '" + userport + "' Please check if the device port is correct or the permissions for port are correct"
+        messagebox.showerror("showerror", ErrorMsg)
+
+#function use to take output from Device on specific button pressed in GUI
+def getOuptutFromPort(command,startRegex,endRegex):
+    PORT=setupPort(port)
+    PORT.write((command).encode())
+    PORT.write(("\n").encode())
+    try:
+        lines = PORT.readlines()
+    except SerialException:
+        ErrorMsg = f"Device disconnected or multiple port access"
+        messagebox.showerror("showerror", ErrorMsg)
+
+    output=""
+    for line in lines:
+        tmp = line.strip().lower()
+        output+=str(line.decode('UTF-8'))
+    if  "Select Test Scenario." in output:
+        PORT.write(("x").encode())
+        PORT.write(("\n").encode())
+        ErrorMsg = f"Terminating Test Mode of MicroController . Please Try again."
+        messagebox.showerror("showerror", ErrorMsg)
+        return
+    if "not registered" in output:
+        ErrorMsg = "'" + command + "'" " Command is not registered\n Enable appropriate command on the device and retry!"
+        messagebox.showerror("showerror", ErrorMsg)
+        return
+
+    file = open(path+"/cmd_output.txt",'w+')
+    file.write(output)
+    file.close()
+    newfile = open(path+'/cmd_output.txt', 'r')
+    if os.path.exists(path+'/cmd_history.txt'):
+        prevfile = open(path+'/cmd_history.txt', 'r')
+    else:
+        prevfile = open(path+'/cmd_history.txt', 'w+')
+    new = ''
+    new += newfile.read()
```

Comment:
Is there any reason why we are writing to cmd_output.txt and then again reading from it?

Reply:
Yes, we have two files:
1) cmd_output stores the output of the last executed command.
2) cmd_history stores the output of all the previously run commands.